.github/actions/detect-workflow/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang@sha256:3b7fa1bed1510cfdadbbd839ab1c2c4ca5ee86bc259d2cb71142f6bd4e70b59f as builder
+
```

Comment:
a little late, but I was wondering if we could avoid dockerfile by using:
```
- uses: action/setup-go@xxx
- uses: action/checkout@xxx
- runs: go build
- runs: run the program
```

I ask because docker build/run take longer, and because it would avoid maintaining the hashes in the docker file.

I don't know if this code works, though.. especially the part on checkout. I feel that it might, the same way that the dockerfile is able to reference its own code...